packages/nc-gui/components/import/JSONImport.vue

```diff
@@ -0,0 +1,479 @@
+              </div>
+            </div>
+          </v-tab-item>
+        </v-tabs>
+
+        <div class="px-4 pb-2">
+          <div class="d-flex">
+            <v-spacer />
+            <span class="caption pointer grey--text" @click="showMore = !showMore">
+              {{ showMore ? $t('general.hideAll') : $t('general.showMore') }}
+              <v-icon small color="grey lighten-1">mdi-menu-{{ showMore ? 'up' : 'down' }}</v-icon>
+            </span>
+          </div>
+          <div class="mb-2 pt-2 nc-json-import-options" :style="{ maxHeight: showMore ? '200px' : '0'}">
+            <p />
+            <!--hint="# of rows to parse to infer data type"-->
+            <v-text-field
+              v-model="parserConfig.maxRowsToParse"
+              style="max-width: 250px"
+              class="caption mx-auto"
+              dense
+              persistent-hint
+              :hint="$t('msg.info.footMsg')"
+              outlined
+              type="number"
+            />
+
+            <v-checkbox
+              v-model="parserConfig.normalizeNested"
+              style="width: 250px"
+              class="mx-auto mb-2"
+              dense
+              hide-details
+            >
+              <template #label>
+                <span class="caption">Flatten nested</span>
+                <v-tooltip bottom position-y="">
+                  <template #activator="{ on }">
+                    <v-icon small class="ml-1" v-on="on">
+                      mdi-information-outline
+                    </v-icon>
+                  </template>
+                  <div class="caption" style="width: 260px">
+                    If flatten nested option is set it will flatten nested object as root level property. In normal case nested object will treat as JSON column.
+                    <br>
+                    <br>
+                    For example the following input: <code class="caption font-weight-bold">{
+                      "prop1": {
+                      "prop2": "value"
+                      },
+                      "prop3": "value",
+                      "prop4": 1
+                      }</code> will treat as:
+                    <code class="caption font-weight-bold">{
+                      "prop1_prop2": "value",
+                      "prop3": "value",
+                      "prop4": 1
+                      }</code>
+                  </div>
+                </v-tooltip>
+              </template>
+            </v-checkbox>
+            <v-checkbox
+              v-model="parserConfig.importData"
+              style="width: 250px"
+              class="mx-auto mb-2"
+              dense
+              hide-details
+            >
+              <template #label>
+                <span class="caption">Import data</span>
+              </template>
+            </v-checkbox>
+          </div>
+        </div>
+      </v-card>
+    </v-dialog>
+
+    <v-tooltip bottom>
+      <template #activator="{on}">
+        <input
```

Comment:
As it's only for JSON, we should only allow json type. Also by doing so, other types will be dimmed so that users can easily select the target file.

```
accept=".json"
```